bundles/binding/org.openhab.binding.ddwrt/src/main/java/org/openhab/binding/ddwrt/internal/ddwrtBinding.java

```diff
@@ -0,0 +1,542 @@
+                }
+            }
+            String username = (String) config.get("username");
+            if (StringUtils.isNotBlank(username)) {
+                ddwrtBinding.username = username;
+            }
+
+            String password = (String) config.get("password");
+            if (StringUtils.isNotBlank(password)) {
+                ddwrtBinding.password = password;
+            }
+
+            String interface_24 = (String) config.get("interface_24");
+            if (StringUtils.isNotBlank(interface_24)) {
+                ddwrtBinding.interface_24 = interface_24;
+            }
+
+            String interface_50 = (String) config.get("interface_50");
+            if (StringUtils.isNotBlank(interface_50)) {
+                ddwrtBinding.interface_50 = interface_50;
+            }
+
+            String interface_guest = (String) config.get("interface_guest");
+            if (StringUtils.isNotBlank(interface_guest)) {
+                ddwrtBinding.interface_guest = interface_guest;
+            }
+
+        }
+    }
+
+    private static class TelnetCommandThread extends Thread {
+
+        private static HashMap<String, String> commandMap = new HashMap<>();
+
+        static {
+            commandMap.put(ddwrtBindingProvider.TYPE_ROUTER_TYPE, "nvram get DD_BOARD");
+            commandMap.put(ddwrtBindingProvider.TYPE_WLAN_24, "ifconfig");
+            commandMap.put(ddwrtBindingProvider.TYPE_WLAN_50, "ifconfig");
+            commandMap.put(ddwrtBindingProvider.TYPE_WLAN_GUEST, "ifconfig");
+        }
+
+        public TelnetCommandThread(String type, Command command) {
+            super();
+            this.type = type;
+            this.command = command;
+        }
+
+        private String type;
+
+        private Command command;
+
+        @Override
+        public void run() {
+            try {
+                TelnetClient client = new TelnetClient();
+                logger.trace("TelnetCommandThread IP ({})", ip);
+                client.connect(ip);
+
+                String state = null;
+                if (command == OnOffType.ON) {
+                    state = "up";
+                } else {
+                    state = "down";
+                }
+
+                String cmdString = null;
+                if (commandMap.containsKey(type)) {
+                    if (type.startsWith("routertype")) {
+                        cmdString = commandMap.get(type);
+                    } else if (type.startsWith("wlan24") && !interface_24.isEmpty()) {
+                        cmdString = commandMap.get(type) + " " + interface_24 + " " + state;
+                    } else if (type.startsWith("wlan50") && !interface_50.isEmpty()) {
+                        cmdString = commandMap.get(type) + " " + interface_50 + " " + state;
+                    } else if (type.startsWith("wlanguest") && !interface_guest.isEmpty()) {
+                        cmdString = commandMap.get(type) + " " + interface_guest + " " + state;
```

Comment:
You should probably use the constants for `routerType`, `wlan24`, etc. you defined in the (to be renamed) `ddwrtBindingProvider` interface.